SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -709,11 +708,11 @@ void USpatialNetDriver::OnLevelAddedToWorld(ULevel* LoadedLevel, UWorld* OwningW
 	{
 		// If load balancing is disabled, we must be the GSM-authoritative worker, so set Role_Authority
 		// otherwise, load balancing is enabled, so check the lb strategy.
-		if (Actor->GetIsReplicated() &&
-			(!bLoadBalancingEnabled || LoadBalanceStrategy->ShouldHaveAuthority(*Actor)))
+		if (Actor->GetIsReplicated())
 		{
-			Actor->Role = ROLE_Authority;
-			Actor->RemoteRole = ROLE_SimulatedProxy;
+			const bool bRoleAuthoritative = !bLoadBalancingEnabled || LoadBalanceStrategy->ShouldHaveAuthority(*Actor);
+			Actor->Role = bRoleAuthoritative ? ROLE_Authority : ROLE_SimulatedProxy;
+			Actor->RemoteRole = bRoleAuthoritative ? ROLE_SimulatedProxy : ROLE_Authority;
```

Comment:
Previously this only functioned correctly if elsewhere in the codebase we set roles to simulated proxy first, we can remove that dependency by setting to simulated proxy here if not set authority